README.md

```diff
@@ -1,8 +1,171 @@
-# ember-modal-dialog
+# Ember Modal Dialog
 
-Component for rendering a modal dialog. Early extraction from a Yapp app. Probably not ready for re-use yet.
+The ember-modal-dialog addon provides components to implement modal dialogs throughout an Ember application using a simple, consistent pattern.
 
-This README outlines the details of collaborating on this Ember addon.
+Unlike other modal libraries for Ember, ember-modal-dialog uses advanced features of HTMLBars to render a modal as a top-level DOM element for layout purposes while retaining its logical position in the Ember view hierarchy. This difference introduces a certain elegance and, dare we say, joy, into the experience of using modals in your app. (For more info on this,
+see the "Wormhole" section below!)
+
+Test examples are located in `tests/dummy/app/templates/application.hbs` and can be run locally by following the instructions in the "Installation" and "Running" sections below.
+
+## Including In An Ember Application
+
+Here is the simplest way to get started with ember-modal-dialog.
+
+```sh
+ember install:addon ember-modal-dialog
+ember install:addon ember-cli-sass
+```
+
+**app.scss**
+```scss
+@import "ember-modal-dialog/ember-modal-structure";
+@import "ember-modal-dialog/ember-modal-appearance";
+```
+
+**application.hbs**
+```htmlbars
+{{#modal-dialog}}
+  Oh hai there!
+{{/modal-dialog}}
+```
+
+## Controller-bound Usage
+
+Here is a more useful example of how to conditionally display a modal based on a user interaction.
+
+**Template**
+
+```htmlbars
+<button {{action "toggleModal"}}>Toggle Modal</button>
+
+{{#if isShowingModal}}
+  {{#modal-dialog alignment='center'
+                  translucentOverlay=true}}
+    Oh hai there!
+  {{/modal-dialog}}
+{{/if}}
+```
+
+**Controller**
+
+```javascript
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  isShowingModal: false,
+  actions: {
+    toggleModal: function(){
+      this.toggleProperty('isShowingModal');
+    }
+  }
+});
+```
+
+## Routable Usage
+
+To assign a modal dialog a dedicated route, place the modal-dialog component directly in a template and transition to the corresponding route.
```

Comment:
I feel like this can be clearer.  Maybe "To have a modal open for a specific route, just drop the `{{modal-dialog}}` into that route's template"